src/open_clip/hf_model.py

```diff
@@ -132,12 +131,21 @@ def __init__(
                 nn.Linear(hidden_size, output_dim, bias=False),
             )
 
-    def forward(self, x: TensorType) -> TensorType:
+    def forward(self, x: TensorType, output_tokens=False) -> TensorType:
         attn_mask = (x != self.config.pad_token_id).long()
         out = self.transformer(input_ids=x, attention_mask=attn_mask)
         pooled_out = self.pooler(out, attn_mask)
+        projected = self.proj(pooled_out)
 
-        return self.proj(pooled_out)
+        if output_tokens:
+            tokens = self.proj(
+                out.last_hidden_state[:, self.pooler.cls_token_position + 1:, :] 
+                if type(self.pooler) == ClsPooler 
+                else out.last_hidden_state
+            )
+            return projected, tokens
```

Comment:
why don't we just do an output dict if we want to add things like this

Reply:
would need to add it to the non HF model too if that is fine I do it

Reply:
ah I see... hmm let me think

Reply:
let's keep it like this for now you're right